src/ray/core_worker/object_interface.cc

```diff
@@ -80,24 +94,107 @@ Status CoreWorkerObjectInterface::Get(
         ids, timeout_ms, worker_context_.GetCurrentTaskID(), &objects));
     RAY_CHECK(ids.size() == objects.size());
     for (size_t i = 0; i < objects.size(); i++) {
-      (*results).emplace(ids[i], objects[i]);
+      if (objects[i] != nullptr) {
+        (*results).emplace(ids[i], objects[i]);
+      }
+    }
+  }
+
+  return Status::OK();
+}
+
+Status CoreWorkerObjectInterface::Wait(const std::vector<ObjectID> &ids, int num_objects,
```

Comment:
Almost all of the code here is duplicated from `Get()` - any simple way to consolidate them?

Reply:
It's indeed similar to some extent, seems like it is not that easy to consolidate because of the differences in logic and structures.